x/emissions/keeper/queryserver/query_server_inferences.go

```diff
@@ -97,10 +100,84 @@ func (qs queryServer) GetLatestNetworkInference(
 		return nil, err
 	}
 
-	return &types.QueryLatestNetworkInferencesAtBlockResponse{
+	return &types.QueryLatestNetworkInferencesResponse{
+		NetworkInferences:         networkInferences,
+		InfererWeights:            synth.ConvertWeightsToArrays(infererWeights),
+		ForecasterWeights:         synth.ConvertWeightsToArrays(forecasterWeights),
+		ForecastImpliedInferences: synth.ConvertForecastImpliedInferencesToArrays(forecastImpliedInferenceByWorker),
+		InferenceBlockHeight:      inferenceBlockHeight,
+		LossBlockHeight:           lossBlockHeight,
+	}, nil
+}
+
+func (qs queryServer) GetLatestAvailableNetworkInference(
+	ctx context.Context,
+	req *types.QueryLatestNetworkInferencesRequest,
+) (
+	*types.QueryLatestNetworkInferencesResponse,
+	error,
+) {
+
+	// Find the latest inference block height
+	_, inferenceBlockHeight, err := qs.k.GetLatestTopicInferences(ctx, req.TopicId)
+	if err != nil {
+		return nil, err
+	}
+	if inferenceBlockHeight == 0 {
+		return nil, status.Errorf(codes.NotFound, "inferences not yet available for topic %v", req.TopicId)
+	}
+
+	topic, err := qs.k.GetTopic(ctx, req.TopicId)
+	if err != nil {
+		return nil, err
+	}
+
+	params, err := qs.k.GetParams(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// calculate the previous loss block height
+	previousLossBlockHeight := inferenceBlockHeight - topic.EpochLength
+
+	for i := 0; i < int(params.DefaultPageLimit); i++ {
+		if previousLossBlockHeight <= 0 {
+			return nil, status.Errorf(codes.NotFound, "losses not yet available for topic %v", req.TopicId)
+		}
+
+		// check for inferences
+		inferences, err := qs.k.GetInferencesAtBlock(ctx, req.TopicId, inferenceBlockHeight)
+		if err == nil && inferences != nil && len(inferences.Inferences) > 0 {
+			// check for losses. If they exist, then break so we can query
+			_, err = qs.k.GetNetworkLossBundleAtBlock(ctx, req.TopicId, previousLossBlockHeight)
+			if err == nil {
+				break
+			}
+		}
+
+		inferenceBlockHeight -= topic.EpochLength
+		previousLossBlockHeight -= topic.EpochLength
+	}
```

Comment:
1. previous block height won't strictly be at that time e.g. if the topic was inactive for a period
2. this is a very inefficient way to discover the right block. we should find an appropriate block and then use functions (GetInferencesAtBlock, GetNetworkLossBundleAtBlock) for their intended purpose given the appropriate blocks

Reply:
Do you have an alternative method in mind for the second point?